File: AllJoynServerExperiences/cs/SampleConfiguration.cs
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

using AllJoynServerExperiences;
using System;
using System.Collections.Generic;
using Windows.UI.Xaml.Controls;

namespace SDKTemplate
{
    public partial class MainPage : Page
    {
        public const string FEATURE_NAME = "AllJoyn Server Experiences";

        List<Scenario> scenarios = new List<Scenario>
        {
            new Scenario() { Title="Secure Server", ClassType=typeof(Scenario1)},
            new Scenario() { Title="Onboarding Producer", ClassType=typeof(Scenario2)}
        };
    }

    public class Scenario
    {
        public string Title { get; set; }
        public Type ClassType { get; set; }
    }
}

File: UserConsentVerifier/cpp/SampleConfiguration.cpp
// Copyright (c) Microsoft. All rights reserved.

#include "pch.h"
#include "MainPage.xaml.h"
#include "SampleConfiguration.h"

using namespace SDKTemplate;

Platform::Array<Scenario>^ MainPage::scenariosInner = ref new Platform::Array<Scenario>
{
    { "Check Consent Availability", "UserConsentVerifier.Scenario1_CheckConsentAvailability" },
    { "Request Consent", "UserConsentVerifier.Scenario2_RequestConsent" }
};
